### DISCUSSION TITLE
use a model variable in an x-blade component dynamically

### DISCUSSION THREAD

**Discussion by kevin4dhd** [livewire/livewire#3200] [1. Help] [answered]
Hi, I don't know if this is possible, but for example, I have the following blade component

![image](https://user-images.githubusercontent.com/22886367/123673010-915a9180-d805-11eb-9f4a-dc88798b8a81.png)

The important thing there is the name of the file, as you can see, I am sending you the model from here.

![image](https://user-images.githubusercontent.com/22886367/123673091-afc08d00-d805-11eb-8658-3e164b8f4d03.png)

So, I want to use that attribute in my blade class, but imagine I have 5 inputs file, as the 5 inputs in the model have getClientOriginalName method, I want to use it in the x-blade component, I have thought of using eval () but I don't know if there is another way.

In summary I want to use the model properties in an x-blade component, I hope you can help me, thanks.

**Comment by joshhanley** [comment, marked as answer]
@kevin4dhd you can access the value of `wire:model` by using `$attributes->wire('model')->value` then use can use that to access that property by name on the LW component using `$this`.

For example in your `inputfile.blade.php`
```blade
@php
$propertyName = $attributes->wire('model')->value;
$property = $this->{$propertyName}
@endphp
```

Hope this helps!

**Reply by kevin4dhd** [reply]
Thank you very much, this is just what I was looking for!

**Reply by joshhanley** [reply]
No worries 🙂
